test/mocks/ImplementationContract.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+/**
+ * @title ImplementationContract
+ * @author pcaversaccio
+ * @dev Allows to mock a simple `payable` implementation contract for an EIP-1167 minimal proxy contract.
+ */
+contract ImplementationContract {
+    bool public isInitialised;
+
+    constructor() payable {}
+
+    /**
+     * @dev An initialisation function that is called once during deployment.
+     */
+    function initialiser() external payable {
+        assert(!isInitialised);
```

Comment:
Custom error works too if you prefer, just makes it easier to debug the line triggering a revert if we have some unique string/error
```suggestion
        require(!isInitialised, "ImplementationContract: already initialized");
```

Reply:
Fair, fixed in 6a79108e628e9539f349410058fa710ea570f640.